scripts/common_cmdline.sh

```diff
@@ -26,19 +26,23 @@ function compileFull()
 {
     local expected_exit_code=0
     local expect_output=0
-    if [[ $1 = '-e' ]]; then
-        expected_exit_code=1
-        expect_output=1
-        shift;
-    fi
-    if [[ $1 = '-w' ]]; then
-        expect_output=1
-        shift;
-    fi
-    if [[ $1 = '-o' ]]; then
-        expect_output=2
-        shift;
-    fi
+
+    case "$1" in
+        '-e')
+            expected_exit_code=1
+            expect_output=1
+            shift;
+            ;;
+        '-w')
+            expect_output=1
+            shift;
+            ;;
+        '-o')
+            expect_output=2
+            shift;
+            ;;
```

Comment:
We should give these options better names. I had to look for calls to `compileFull` to understand what they really mean.
```suggestion
    local expect_output='none'
    case "$1" in
        '--expect-errors')
            expected_exit_code=1
            expect_output='warnings-or-errors'
            shift;
            ;;
        '--expect-warnings')
            expect_output='warnings-or-errors'
            shift;
            ;;
        '--ignore-warnings')
            expect_output='any'
            shift;
            ;;
```